File: src/c#/CharacterRepetition.cs
using System;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine("Ingrese una Cadena: \n");
        string input = Console.ReadLine();

        Dictionary<char, int> uniqueLettersMap = new Dictionary<char, int>();

        foreach (char c in input)
        {
            if (uniqueLettersMap.ContainsKey(c))
            {
                uniqueLettersMap[c]++;
            }
            else
            {
                uniqueLettersMap.Add(c, 1);
            }
        }

        foreach (KeyValuePair<char, int> letterRepetition in uniqueLettersMap)
        {
            Console.WriteLine(
                "La Letra {0} se repite {1} vez/veces.",
                letterRepetition.Key,
                letterRepetition.Value
            );
        }
    }
}

File: actividad2/python/character_repetition.py
def getUniqueLettersWithRepetition(myString):
    uniqueLetters = {}

    for i in myString:
        if not i in uniqueLetters:
            uniqueLetters[i] = 1
        else:
            uniqueLetters[i] = uniqueLetters[i]+1

    return uniqueLetters

def main():
    myString = input("Ingrese una cadena: ")
    print(myString)

    uniqueLetters = getUniqueLettersWithRepetition(myString)

    for i in uniqueLetters:
        print('La letra', i, 'se repite', uniqueLetters[i], "vez/veces.")


main()
